src/Datadog.Trace/CorrelationIdentifier.cs

```diff
@@ -57,11 +59,11 @@ public static string Env
         /// Gets the id of the active trace.
         /// </summary>
         /// <returns>The id of the active trace. If there is no active trace, returns zero.</returns>
-        public static ulong TraceId
+        public static TraceId? TraceId
```

Comment:
@zacharycmontoya won't such change break similarly to: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/82#discussion_r594488610

Reply:
Yes, a lot of these changes modify the signatures of API's to use this new type instead of long's. I'll review the PR in its entirety so I can propose a more helpful suggestion, but we wouldn't be able to accept many of the internal API's that currently exist in the PR

Reply:
Disregarding the question of API changes, this could be a non-nullable `TraceId` where the default value is `TraceId.Zero`, since that would more closely align with the previous default value